CRM/Upgrade/Incremental/php/FourSeven.php

```diff
@@ -421,6 +421,90 @@ public function upgrade_4_7_24($rev) {
     $this->addTask('CRM-20958 - Add modified_date to civicrm_case', 'addColumn',
       'civicrm_case', 'modified_date', "timestamp NULL  DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP COMMENT 'When was the case (or closely related entity) was created or modified or deleted.'");
 
+    $this->addTask('CRM-20958 - Suspend modification tracking during upgrade', 'toggleModificationTracking', FALSE);
+
+    list($minId, $maxId) = CRM_Core_DAO::executeQuery(
+      "SELECT coalesce(min(id),0), coalesce(max(id),0) FROM civicrm_activity"
+    )->getDatabaseResult()->fetchRow();
+    for ($startId = $minId; $startId <= $maxId; $startId += self::BATCH_SIZE) {
+      $endId = $startId + self::BATCH_SIZE - 1;
+      $vars = array(1 => array($startId, 'Int'), 2 => array($endId, 'Int'));
+
+      $title = sprintf('CRM-20958 - Compute civicrm_activity.created_date from civicrm_log (%d => %d)', $startId, $endId);
+      $sql = 'UPDATE civicrm_activity
+       SET created_date = (SELECT MIN(l.modified_date) FROM civicrm_log l WHERE l.entity_table ="civicrm_activity" AND civicrm_activity.id = l.entity_id)
```

Comment:
One thing I'm not certain about... this code initializes `civicrm_activity.created_date` (etal) using `civicrm_log.modified_date`, but they have different data-types (`TIMESTAMP` vs `DATETIME`, respectively).

@eileenmcnaughton @seamuslee001 To my recollection, `DATETIME` and `TIMESTAMP` handle timezones differently, so wondering if there's some kind of filtering I should do here to adjust for timezone.

However, my gut says that storing `civicrm_log.modified_date` as `DATETIME` is actually a subtle dirtiness/corruption. Ex: [This](https://github.com/civicrm/civicrm-core/blob/4.7.22/CRM/Activity/BAO/Activity.php#L633-L640) suggests that `civicrm_log.modified_date` would be initialized based on `date('YmdHis')` (ie the user's *perceived timestamp* based on active PHP timezone setting) rather than a *canonical timestamp*. But different `civicrm_log` records can be created by different people/sessions/timezones, so the raw data in `civicrm_log` wouldn't be reliably articulated in any one timezone.

This wouldn't really the fault of my issue/patch -- if the theory is true, you'd expect to see idiosyncrasies in the display of this data. For migration purposes, somewhere you'd have to bite the bullet and accept that timestamps would be wrong +/- a few hours?